OpenRA.Mods.Common/Activities/DeliverUnit.cs

```diff
@@ -46,7 +46,7 @@ protected override void OnFirstRun(Actor self)
 			// In case this activity was queued (either via queued order of via AutoCarryall)
 			// something might have happened to the cargo in the time between the activity being
 			// queued and being run, so short out if it is no longer valid.
-			if (carryall.Carryable == null)
+			if (carryall.Carryable == null || carryall.State != Carryall.CarryallState.Carrying)
```

Comment:
The code that changes the state also sets Carryable to null. So this doesn't seem like the right fix? Did you try to run this through the replay?

Reply:
I did, yes. The carryable was reserved for pickup (`CarryallState.Reserved`), but not actually picked when this action runs.

Reply:
I see. And I assume the rest of the code is able to handle this case?